lib/ipmi_hpeoem.c

```diff
@@ -0,0 +1,428 @@
+/*
+ * Copyright (c) 2021, Jiajie Chen
+ * All rights reserved.
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * - Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of Dell Inc nor the names of its contributors
+ * may be used to endorse or promote products derived from this software 
+ * without specific prior written permission.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ Reference doc:
+ HPE iLO IPMI User Guide
+ https://support.hpe.com/hpesc/public/docDisplay?docId=c04530505&docLocale=en_US
+ */
+
+#include <ipmitool/ipmi.h>
+#include <ipmitool/ipmi_intf.h>
+#include <ipmitool/ipmi_lanp.h>
+#include <ipmitool/ipmi_hpeoem.h>
+#include <ipmitool/log.h>
+
+static int current_arg = 0;
+
+static void usage(void);
+static void ipmi_hpeoem_lan_usage(void);
+static int ipmi_hpeoem_lan_get_nic_selection(struct ipmi_intf *intf, int chan);
+static int ipmi_hpeoem_parse_nic_selection_mode(char **argv);
+static int ipmi_hpeoem_lan_set_nic_selection(struct ipmi_intf *intf, int channel, int nic_selection);
+
+// 224 iLO Network Configuration Status and NIC Selection
+// data 3
+char ILONICSelectionMode[2][50] = {
+    "dedicated",
+    "shared",
+};
+
+// 197 iLO Shared NIC Selection and Shared NIC Port Number
+// data 1
+char ILOSharedNICSelection[3][50] = {
+    "Unknown",
+    "LOM",
+    "ALOM",
+};
+
+// 197 iLO Shared NIC Selection and Shared NIC Port Number
+// data 2
+char ILOSharedNICPort[3][50] = {
+    "Unknown",
+    "NIC port 1",
+    "NIC port 2",
+};
+
+#define INVALID -1
+#define DEDICATED 0
+#define SHARED_WITH_LOM1 1
+#define SHARED_WITH_LOM2 2
+#define SHARED_WITH_ALOM1 3
+#define SHARED_WITH_ALOM2 4
```

Comment:
I'd suggest using an `enum` as it is better for debugging. Enums are displayed as names in GDB, macros aren't.